app/api/events.py

```diff
@@ -93,9 +93,28 @@ def get_upcoming_events(request: Request, token: AccessTokenPayload = Depends(op
     return [Event.model_validate(event) for event in upcoming_events]
 
 
+@router.get('/past-events/count')
+def get_past_events_count(request: Request, token: AccessTokenPayload = Depends(optional_authentication)):
+    """Get count of past events"""
+    db = get_database(request)
+
+    # Get current UTC datetime
+    current_datetime = datetime.now(timezone.utc)
+
+    # Apply search filter according to role
+    search_filter = {"date": {"$lt": current_datetime}}
+    if token and token.role != Role.admin:
+        search_filter["public"] = True
```

Comment:
I'm not sure this is leads to a valid mongodb query. You can copy the search filter logic from the `get_past_events` endpoint instead